src/ThreeEditor/js/viewport/Viewport.ClippedView2.js

```diff
@@ -0,0 +1,214 @@
+import * as THREE from 'three';
+import { GUI } from 'three/examples/jsm/libs/lil-gui.module.min.js';
+import { CSG } from '../libs/csg/csg-lib';
+import { debounce } from 'throttle-debounce';
+
+/*
+This function is responsible for preparing scene with clipping plane and stencil materials
+It creates and manages controls for shifting clipping planes in X,Y,Z axis.
+Stencil materials are used as a trick in order to avoid display of transparent inner walls of clipped objects (zones/bodies).
+We aim at assigning stencil materials to hold different colors than the walls of the objects and represent the internal material which is filling the zone.
+This is especially useful for such clipped views where at the same time internal material and wall of the object is visible at same time
+(for example torus clipped by a plane containing its axis of symmetry, stencil material fills two full circles, while wall material is drawn on the visible outer wall).
+ */
+export function ViewportClippedView2(
+	name,
+	editor,
+	viewport,
+	planeHelpers,
+	initialObjects,
+	signalGeometryChanged,
+	signalGeometryAdded,
+	signalGeometryRemoved,
+	container,
+	{ clipPlane, planeHelperColor, planePosLabel }
+) {
+	this.name = name;
+	this.editor = editor;
+
+	// default order is zero, we assign higher order to stencil plane to display it in front of other objects
+	const STENCIL_RENDER_ORDER = 1;
+
+	// several objects we are using (stencil plan) are limited to a square on a plane with given size (side length)
+	// it limits range of applicability of clipping plane
+	// it also limits the size (horizontally and vertically) of stencil plane
+	// here side length is set to 10 cm
+	const CLIPPING_SIZE = 200;
```

Comment:
Will this work with a 40x40x40 cm cube ?

Reply:
What about this project https://gist.githubusercontent.com/grzanka/cd47cacce2f6d5986397630d76580d11/raw/e6ca6a23fbae78b04ddcfaf2582bc7c3f972e995/2022let_plan2.json ?

Reply:
I think yes